andrewballantyne/odh-dashboard
# Enable Nvidia NIM as an application

## Description



Create an OdhApplication CR to add a tile for the Nvidia NIM application.
Support enabling the NIM application with a Nvidia ngc api key.
Create a CronJob to validate the api key and enable the NIM application.

<img width="1712" alt="Screenshot 2024-06-25 at 5 06 16 PM" src="https://github.com/andrewballantyne/odh-dashboard/assets/42279951/d986da12-b0b5-41ac-80e8-4f6c87ff30dd">


## How Has This Been Tested?




Negative scenario:
1. Enter an invalid ngc api key 
2. Check if the Job for validating the api key fails
3. Check if the NIM application is not enabled
4. Check if the CronJob (executed daily) for validating the api key fails
5. Check if the Secret for pulling NIM images is not created
6. Check if the ConfigMap that stores NIM images data is not created

Positive scenario:
1. Enter a valid ngc api key 
2. Check if the Job for validating the api key succeeds
3. Check if the NIM application is enabled
4. Check if the CronJob (executed daily) for validating the api key succeeds
5. Check if the Secret for pulling NIM images is created correctly
6. Check if the ConfigMap that stores NIM images data is created correctly

## Test Impact



No impact on existing code

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
